apps/web/src/components/borrow/table/DepositModal.tsx

```diff
@@ -0,0 +1,293 @@
+        currentAccount,
+      )
+      const isDeposited = await response.wait()
+      setDepositCompleted(!!isDeposited)
+      resetState()
+    } catch (error) {
+      setHasError(error)
+    } finally {
+      setLoading(false)
+    }
+  }
+  const Title = () => (
+    <Text color="oxfordBlue" type="heading 2" as="span">
+      Deposit asset
+    </Text>
+  )
+
+  const transactionData = [
+    {
+      name: 'Supply APY',
+    },
+    {
+      name: 'Health factor',
+    },
+  ]
+  const handleInputCHange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = Number(e.target.value)
+    if (value > Number(data.balance) || value === 0) {
+      setAmountError(true)
+    } else {
+      setAmountError(false)
+    }
+    setAmount(value)
+  }
+  const tableColumns = {
+    action: {
+      header: 'Amount',
+      Cell: () => (
+        <ColumnLayout gap={0.5}>
+          <StackLayout>
+            <BaseInput
+              status={amountError ? 'error' : undefined}
+              onChange={handleInputCHange}
+              value={amount}
+            />
+          </StackLayout>
+          <Button
+            appearance="secondary"
+            variant="thin"
+            size="small"
+            className="u-fit-content-width"
+            onClick={() => setAmount(Number(data.balance))}
+          >
+            MAX
+          </Button>
+        </ColumnLayout>
+      ),
+      alignText: 'left',
+    },
+    asset: {
+      header: '',
+      Cell: ({ data: { symbol } }: CellProps) => <span>{symbol}</span>,
+      alignText: 'right',
+    },
+  }
+
+  const getValue = (index: number) => {
+    switch (index) {
+      case 0:
+        return (
+          <ColumnLayout align="flex-end" center>
+            <Text as="span" type="body 1" color="oxfordBlue">
+              <FormattedNumber value={data.supplyAPY} percent />
+            </Text>
+          </ColumnLayout>
+        )
+      case 1:
+        return (
+          <StackLayout>
+            <ColumnLayout align="flex-end" center>
+              <Icon name="union" size={16} color={themeContext.buttonDark} />
+              <Icon
+                name="arrow-right"
+                size={15}
+                color={themeContext.buttonDark}
+              />
+              <Text as="span" type="body 1" color="oxfordBlue">
+                1.00
+              </Text>
+            </ColumnLayout>
+            <Text as="span" type="body 1" color="oxfordBlue">
+              {'Liquidation at < 1.00'}
+            </Text>
+          </StackLayout>
+        )
+    }
+  }
+
+  const transactionColumns = {
+    action: {
+      header: 'Transaction overview',
+      Cell: ({ data: { name } }: CellProps) => (
+        <Text type="body 3" color="oxfordBlue" as="span">
+          {name}
+        </Text>
+      ),
+      alignText: 'left',
+    },
+    value: {
+      header: '',
+      Cell: ({ index }: CellProps) => getValue(index),
+      alignText: 'right',
+    },
+  }
```

Comment:
let's try what I did recently I think I added here
`as Record<string, ColumnData<WalletBalance>`
something like this, and then you can remove CellProps assertion.

Reply:
That is giving a more correct type, because CellProps is `{[string]: unknown}`